pychromecast/quick_play.py

```diff
@@ -0,0 +1,43 @@
+""" Choose a controller and quick play """
+
+from .controllers.youtube import YouTubeController
+from .controllers.supla import SuplaController
+
+
+def quick_play(cast, app_name, data):
+    """
+    CAST_APP_SCHEMA = {
+        vol.Required('app_name', default=""): cv.string,
+        vol.Required('data'): vol.Schema({
+            vol.Required("media_id"): cv.string,
+            vol.Optional("media_type"): cv.string,
+            vol.Optional("enqueue", default=False): cv.boolean,
+            vol.Optional("index"): cv.string,
+            vol.Optional("extra1"): cv.string,
+            vol.Optional("extra2"): cv.string,
+        }),
+    }
+    """
+
+    if app_name == "youtube":
+        controller = YouTubeController()
+        kwargs = {
+            "video_id": data.pop("media_id"),
+            "enqueue": data.pop("enqueue", False),
+            "playlist_id": data.pop("extra1", None),
+        }
+    elif app_name == "supla":
+        controller = SuplaController()
+        kwargs = {
+            "media_id": data.pop("media_id"),
+            "is_live": data.pop("extra2", None),
+        }
+    else:
+        raise NotImplementedError()
+
+    cast.wait()
```

Comment:
We should not do a `cast.wait()` here:
- It's not suitable for Home Assistant because it blocks
- It's not compatible with running the SocketClient in a custom event loop instead of with its own thread.

Instead, the caller of `quick_play` should ensure the Chromecast object is in the correct state.

Reply:
Could I use an app launch callback here instead?

Reply:
That should not be needed, `BaseController.send_message()` will automatically launch the necessary app and wait for it to be launched before sending the message:
https://github.com/home-assistant-libs/pychromecast/blob/7372f8d24209aad9b5f7f61f5988f28994a7ec60/pychromecast/controllers/__init__.py#L76-L81